models/forms/registration.go

```diff
@@ -0,0 +1,82 @@
+package forms
+
+import (
+	"log"
+	"strings"
+
+	"go-google-scraper-challenge/helpers"
+	"go-google-scraper-challenge/models"
+
+	"github.com/beego/beego/v2/core/validation"
+)
+
+type RegistrationForm struct {
+	Email                string `valid:"Email; Required"`
+	Password             string `valid:"Required; MinSize(6)"`
+	PasswordConfirmation string `valid:"Required; MinSize(6)"`
+}
+
+func init() {
+	lowerCaseErrorMessage := map[string]string{}
+	for key, value := range validation.MessageTmpls {
+		lowerCaseErrorMessage[key] = strings.ToLower(value)
+	}
+
+	validation.SetDefaultMessage(lowerCaseErrorMessage)
+}
+
+// Valid adds custom validation to registration form, sets error when the validation failed.
+func (registrationForm *RegistrationForm) Valid(v *validation.Validation) {
+	// This will raise an error if user with the given email does not exist.
+	_, err := models.GetUserByEmail(registrationForm.Email)
+	if err == nil {
```

Comment:
I am not sure about the check on error here as it would make more sense to check that the user is not equal to nil:

```suggestion
	if user != nil {
```

Checking on `err` makes if confusing

Reply:
@olivierobert I just found that they got the `.Exist()` function so I use it on the model instead of `.Once()`. It will return a boolean so we don't have to handle the error and also avoid confusion. The change is in [13e5f39](https://github.com/carryall/go-google-scraper-challenge/pull/21/commits/13e5f393083ab72871e3469aefcbe8917f285075)
```
func UserEmailAlreadyExist(email string) (userExist bool) {
	ormer := orm.NewOrm()

	return ormer.QueryTable(User{}).Filter("Email", email).RelatedSel().Exist()
}
```